server.go

```diff
@@ -318,6 +297,53 @@ func (a *App) reply(w http.ResponseWriter, r *http.Request) {
 	}
 	successReply(w)
 }
+
+func (a *App) run(w http.ResponseWriter, r *http.Request) {
+	var msg Message
+	if err := json.NewDecoder(r.Body).Decode(&msg); err != nil {
+		errorReply(w, http.StatusBadRequest, "couldn't parse json")
+		return
+	}
+
+	msg.TwinSrc = a.twin
+	msg.Retqueue = uuid.New().String()
+
+	// if len(msg.TwinDst) > 1 || msg.TwinDst[0] != a.twin {
+	// 	errorReply(w, http.StatusBadRequest, "couldn't parse json")
+	// 	return
+	// }
+
+	err := a.backend.PushToLocal(r.Context(), msg)
+	fmt.Println(err)
+	if err != nil {
+		errorReply(w, http.StatusInternalServerError, err.Error())
+		return
+	}
+
+	ret := MessageIdentifier{
+		Retqueue: msg.Retqueue,
+	}
+
+	json.NewEncoder(w).Encode(&ret)
+}
+
+func (a *App) getResult(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	var msgIdentifier MessageIdentifier
+	if err := json.NewDecoder(r.Body).Decode(&msgIdentifier); err != nil {
+		errorReply(w, http.StatusBadRequest, "couldn't parse json")
+		return
+	}
+
+	response, err := a.backend.GetMessageReply(r.Context(), msgIdentifier)
+	if err != nil {
+		errorReply(w, http.StatusInternalServerError, err.Error())
```

Comment:
It's not always an internal server error. The error can happen if redis is down (which should be an internal server error), the error entered a non-valid return queue or a not-filled return queue which resulted it being empty (which should be NotFound).